src/tester.rs

```diff
@@ -301,25 +301,39 @@ impl<T: Testable,
             self_: fn($($name),*) -> T,
             a: ($($name,)*),
         ) -> Option<TestResult> {
-            for t in a.shrink() {
-                let ($($name,)*) = t.clone();
+            let mut values = a.shrink().collect::<Vec<_>>();
```

Comment:
Is it feasible to collect all shrinks? I'm thinking it should be possible to feed shrink a very very large list.